src/nodeBuffer.ml

```diff
@@ -44,29 +45,341 @@ external allocFillStringWithEncoding: int -> fill:Js.String.t ->
+
 external byteLengthString : Js.String.t -> int = "Buffer.byteLength" [@@bs.val]
 external byteLengthStringWithEncoding : Js.String.t -> encoding:Js.String.t -> int =
   "Buffer.byteLength" [@@bs.val]
 external byteLengthBuffer : t -> int = "Buffer.byteLength" [@@bs.val]
-external byteLengthInt8Array : int Int8Array.typed_array -> int =
+external byteLengthInt8Array : Int8Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthUint8Array : int Uint8Array.typed_array -> int =
+external byteLengthUint8Array : Uint8Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthInt16Array : int Int16Array.typed_array -> int =
+external byteLengthInt16Array : Int16Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthUint16Array : int Uint16Array.typed_array -> int =
+external byteLengthUint16Array : Uint16Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthInt32Array : int Int32Array.typed_array -> int =
+external byteLengthInt32Array : Int32Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthUint32Array : int Uint32Array.typed_array -> int =
+external byteLengthUint32Array : Uint32Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthFloat32Array : float Float32Array.typed_array -> int =
+external byteLengthFloat32Array : Float32Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthFloat64Array : float Float64Array.typed_array -> int =
+external byteLengthFloat64Array : Float64Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
 external byteLengthDataView : DataView.t -> int = "Buffer.byteLength" [@@bs.val]
 external byteLengthArrayBuffer : ArrayBuffer.t -> int = "Buffer.byteLength" [@@bs.val]
+
 external compare : t -> t -> int = "Buffer.compare" [@@bs.val]
+
+external concat : t array -> t = "Buffer.concat" [@@bs.val]
+external concatLength : t array -> length:int -> t = "Buffer.concat" [@@bs.val]
+
+external isEncoding : Js.String.t -> bool = "Buffer.isEncoding" [@@bs.val]
+
+type buffer
+external buffer : buffer = "Buffer" [@@bs.val]
+external poolSize : int = "Buffer.poolSize" [@@bs.val]
+external setPoolSize : buffer -> int -> int = "poolSize" [@@bs.set]
+let setPoolSize n = setPoolSize buffer n
+
+external copy : t -> t -> int = "" [@@bs.send]
+external copyOffset : t -> t -> targetStart:int -> int = "copy" [@@bs.send]
+external copyOffsetSource : t -> t -> targetStart:int -> sourceStart:int ->
+  int = "copy" [@@bs.send]
+external copyRangeSource : t -> t -> targetStart:int -> sourceStart:int ->
+  sourceEnd:int -> int = "copy" [@@bs.send]
+external copyToUint8Array : t -> Uint8Array.t -> int = "copy" [@@bs.send]
+external copyToUint8ArrayOffset : t -> Uint8Array.t -> targetStart:int ->
+  int = "copy" [@@bs.send]
+external copyToUint8ArrayOffsetSource : t -> Uint8Array.t -> targetStart:int ->
+  sourceStart:int -> int = "copy" [@@bs.send]
+external copyToUint8ArrayRangeSource : t -> Uint8Array.t -> targetStart:int ->
```

Comment:
Yeah, casting optional args into name suffix is a pain. Could we start a wiki with some conventions? Meanwhile will apply your suggestions, they are reasonable.